### DISCUSSION TITLE
Is a queued Sidekiq::Batch::Empty a sign that we didn't use batches corectly?

### DISCUSSION THREAD

**Discussion by bmulholland** [sidekiq/sidekiq#6139] [Q&A] [answered]
Is `Sidekiq::Batch::Empty` scheduled for every batch, or only for batches that don't have any jobs added?

I'm seeing some of those jobs in our queue and wondering if that's a sign of a bug in our code to investigate.

Thank you!

**Comment by mperham** [comment, marked as answer]
Yeah, it's a signal someone created a batch with no jobs in it. Technically not a great idea but the Empty job is there to ensure any callbacks are fired.

**Reply by bmulholland** [reply]
Got it, thanks. It's a good symptom to observe: all our batches should follow the recommendations to never be empty. We'll track it down.

**Reply by bmulholland** [reply]
Appreciate the fast response!
